tests/vectors_sig.c

```diff
@@ -165,109 +212,340 @@ OQS_STATUS sig_vector(const char *method_name,
+	}
+
+
+	fprintBstr(fh, "testPassed: ", (const uint8_t *)&testPassed, 1);
+
+	goto cleanup;
+
+err:
+	ret = EXIT_FAILURE;
+	goto cleanup;
+
+algo_not_enabled:
+	ret = EXIT_SUCCESS;
+
+cleanup:
+	OQS_SIG_free(sig);
+	return ret;
+
+}
+
+static int sig_gen_vector(const char *method_name,
+                          uint8_t *prng_output_stream,
+                          const uint8_t *sigGen_sk, const uint8_t *sigGen_msg, size_t sigGen_msgLen, const uint8_t *sigGen_sig, int randomized) {
+
+	FILE *fh = NULL;
+	uint8_t *signature = NULL;
+	OQS_SIG *sig = NULL;
+	int rc = -1, ret = -1;
+	size_t sigLen;
+
+	sig = OQS_SIG_new(method_name);
+	if (sig == NULL) {
+		printf("[vectors_sig] %s was not enabled at compile-time.\n", method_name);
+		goto algo_not_enabled;
+	}
+
+	sigLen = sig->length_signature;
+
+	fh = stdout;
+
+	signature = malloc(sigLen);
+
+	if (signature == NULL) {
+		fprintf(stderr, "[test_acvp_sig] %s ERROR: malloc failed!\n", method_name);
+		goto err;
+	}
+
+	if ((randomized && prng_output_stream == NULL) || (sigGen_sk == NULL) || (sigGen_msg == NULL) || (sigGen_sig == NULL)) {
+		fprintf(stderr, "[test_acvp_sig] %s ERROR: inputs NULL!\n", method_name);
+		goto err;
+	}
+
+	if (!strcmp(method_name, "ML-DSA-44")) {
+#ifdef OQS_ENABLE_SIG_ml_dsa_44
+		rc = pqcrystals_ml_dsa_44_ref_signature_internal(signature, &sigLen, sigGen_msg, sigGen_msgLen, NULL, 0, prng_output_stream, sigGen_sk);
+#endif
+	} else if (!strcmp(method_name, "ML-DSA-65")) {
+#ifdef OQS_ENABLE_SIG_ml_dsa_65
+		rc = pqcrystals_ml_dsa_65_ref_signature_internal(signature, &sigLen, sigGen_msg, sigGen_msgLen, NULL, 0, prng_output_stream, sigGen_sk);
+#endif
+	} else if (!strcmp(method_name, "ML-DSA-87")) {
+#ifdef OQS_ENABLE_SIG_ml_dsa_87
+		rc = pqcrystals_ml_dsa_87_ref_signature_internal(signature, &sigLen, sigGen_msg, sigGen_msgLen, NULL, 0, prng_output_stream, sigGen_sk);
+#endif
+	} else {
+		goto err;
+	}
+
+	if (rc) {
+		fprintf(stderr, "[test_acvp_sig] %s ERROR: mlca_sig_sign_internal failed!\n", method_name);
+		goto err;
+	}
+	fprintBstr(fh, "signature: ", signature, sig->length_public_key);
+
+	if (!memcmp(signature, sigGen_sig, sigLen)) {
+		ret = EXIT_SUCCESS;
+	} else {
+		ret = EXIT_FAILURE;
+		fprintf(stderr, "[test_acvp_sig] %s ERROR: public key or private key doesn't match!\n", method_name);
+	}
+	goto cleanup;
+
+err:
+	ret = EXIT_FAILURE;
+	goto cleanup;
+
+algo_not_enabled:
+	ret = EXIT_SUCCESS;
+
+cleanup:
+	OQS_MEM_insecure_free(signature);
+	OQS_SIG_free(sig);
+	return ret;
+}
+#endif
+
 int main(int argc, char **argv) {
-	OQS_STATUS rc;
+	OQS_STATUS rc = OQS_SUCCESS;
 
 	OQS_init();
 
-	if (argc != 8) {
-		fprintf(stderr, "Usage: vectors_sig algname prng_output_stream sig_msg sig_sk verif_sig verif_pk verif_msg\n");
-		fprintf(stderr, "  algname: ");
-		for (size_t i = 0; i < OQS_SIG_algs_length; i++) {
-			if (i > 0) {
-				fprintf(stderr, ", ");
-			}
-			fprintf(stderr, "%s", OQS_SIG_alg_identifier(i));
-		}
+	if (argc == 222) {
```

Comment:
Where does 222 come from?

Reply:
Was a placeholder for a better check :) Thanks for the catch! Updated in ed194b7dbad65ed5bc795eb515833901b1bfe304